### PR TITLE
[Frontend] Add `logits_processors` as an extra completion argument

### PR DESCRIPTION
**Pull request by bradhilton**
Adds `logits_processors` as an extra (optional) argument to `ChatCompletionRequest` and `CompletionRequest`:

```python
    logits_processors: Optional[List[Union[str, LogitsProcessorConstructor]]] = Field(
        default=None,
        description=(
            "A list of either qualified names of logits processors, or "
            "constructor objects, to apply when sampling. A constructor is "
            "a JSON object with a required 'qualname' field specifying the "
            "qualified name of the processor class/factory, and optional "
            "'args' and 'kwargs' fields containing positional and keyword "
            "arguments. For example: {'qualname': 'my_module.MyLogitsProcessor',"
            "'args': [1, 2], 'kwargs': {'param': 'value'}}."))
 ```
 
 This enables passing built-in or custom logits processors to direct the sampling process.

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Hmm, this seems a bit dangerous as it allows the user to execute any Python module server-side. A better solution would be to introduce a registry of logits processors which the user can reference by their key in the registry.

**Comment by bradhilton** [comment]
> Hmm, this seems a bit dangerous as it allows the client to execute any Python module server-side. A better solution would be to introduce a registry of logits processors which the client can reference by their key in the registry.

Great feedback, thank you @DarkLight1337. I added an optional `--logits-processor-pattern` engine argument to only whitelist matching processor qualified names. It's `None` by default, disallowing the usage of the `logits_processors` argument entirely.

File: vllm/config.py [hunk 1]
```diff
@@ -156,41 +156,45 @@ class ModelConfig:
...
+                 limit_mm_per_prompt: Optional[Mapping[str, int]] = None,
+                 use_async_output_proc: bool = True,
+                 config_format: ConfigFormat = ConfigFormat.AUTO,
+                 hf_overrides: Optional[HfOverrides] = None,
+                 mm_processor_kwargs: Optional[Dict[str, Any]] = None,
+                 mm_cache_preprocessor: bool = False,
+                 override_neuron_config: Optional[Dict[str, Any]] = None,
+                 override_pooler_config: Optional["PoolerConfig"] = None,
+                 logits_processor_pattern: Optional[str] = None) -> None:
         self.model = model
```
Comment (bradhilton) [inline vllm/config.py:198]: `format.sh` decided to reformat this initializer 🤷‍♂️

**Review by DarkLight1337 — APPROVED** [review]
Using a pattern to restrict access also works, thanks for adding this!

**Comment by bradhilton** [comment]
@DarkLight1337 `buildkite/ci/pr/multi-step-tests-4-gpus` failed due to a connect timeout. Is this normal test flakiness or cause for concern?

**Comment by DarkLight1337** [comment]
Let me retry the test.

**Comment by bradhilton** [comment]
Thanks @DarkLight1337 !

**Comment by xu-song** [comment]
`tokenizer` is commonly used in logits processor, but it can not be passed as `args` to `logits_processors`
